composer/core/state.py

```diff
@@ -47,6 +49,32 @@ def null(precision):
         return null
 
 
+def _ensure_backwards_compatible_checkpointing(state_dict: types.StateDict):
```

Comment:
Who is the backwards compatible checkpointing for?

Is there a plan to eventually remove this function?

Reply:
I figured we'll need this as we tweak the checkpoint format. We may remove parts of it as checkpoints become more and more outdated as composer updates.